src/App.tsx

```diff
@@ -3,12 +1,34 @@
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { SelectedType } from './types/SelectedType';
+
+import { getFilteredData } from './utils/getFilteredData';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [query, setQuery] = useState('');
+  const [selectedCondition, setSelectedCondition] = useState<SelectedType>(
+    SelectedType.ALL,
+  );
+
+  useEffect(() => {
+    setLoading(true);
+
+    getTodos()
+      .then(setTodos)
+      .finally(() => setLoading(false));
+  }, []);
+
+  const filteredData = getFilteredData(todos, { query, selectedCondition });
```

Comment:
You're directly manipulating the DOM by calling `getFilteredData`. This goes against the React philosophy of not directly manipulating the DOM and instead using the React state and lifecycle methods. Consider moving this logic inside a `useEffect` hook and storing the result in a state variable.